### PR TITLE
Fixconvert #195

### PR DESCRIPTION
**Pull request by pchalamet**

### CODE REVIEW

**Comment by pchalamet** [comment]
Anthology is wrong for cassandra-sharp-contrib:

{code}
apps: []
projects:
  - fx:
      version: v4.5
      profile: null
      identifier: null
    guid: 9E8648A4-D25A-4CFA-AAEE-20D9D63FF571
    file: cassandra-sharp-contrib\CassandraSharp.Contrib.log4netUnitTests\CassandraSharp.Contrib.log4netUnitTests-net45.csproj
    out: cassandrasharp.contrib.log4netunittests.dll
    tests: true
    assemblies:
      - assembly: system
      - assembly: system.core
    packages:
      - package: cassandra-sharp
      - package: log4net
      - package: nunit
      - package: system.reactive
      - package: system.reactive.core
      - package: system.reactive.interfaces
      - package: system.reactive.linq
      - package: system.reactive.platformservices
    projects:
      - project: cassandrasharp
      - project: cassandrasharp.interfaces
  - fx:
      version: v4.5
      profile: null
      identifier: null
    guid: 925833ED-8653-4E90-9C37-B5B6CB693CF4
    file: cassandra-sharp-contrib\CassandraSharp.Contrib.log4net\CassandraSharp.Contrib.log4net-net45.csproj
    out: cassandrasharp.contrib.log4net.dll
    tests: false
    assemblies:
      - assembly: system
    packages:
      - package: log4net
      - package: system.reactive
      - package: system.reactive.core
      - package: system.reactive.interfaces
      - package: system.reactive.linq
      - package: system.reactive.platformservices
    projects:
      - project: cassandrasharp.interfaces
{code}

**Comment by pchalamet** [comment]
:x: [Build full-build 4.3.50 failed](https://ci.appveyor.com/project/full-build/full-build-shu78/build/4.3.50) (commit https://github.com/full-build/full-build/commit/307acb2965 by @)

File: src/FullBuild/Parsers/PackageRelationship.fs [hunk 1]
```diff
@@ -32,13 +33,30 @@ let GetFrameworkDependencies (xnuspec : XDocument) =
 let GetPackageDependencies (xnuspec : XDocument) =
```
Comment (pchalamet) [inline src/FullBuild/Parsers/PackageRelationship.fs]: Don't know what it does now but it does it wrongly for sure.

File: src/FullBuild/Parsers/PackageRelationship.fs [hunk 2]
```diff
@@ -49,13 +67,13 @@ let rec BuildPackageDependencies (packages : PackageId seq) =
             let nuspecFile = pkgDir |> GetFile (FsHelpers.AddExt NuSpec (package.toString))
             let xnuspec = XDocument.Load (nuspecFile.FullName)
             let dependencies = GetPackageDependencies xnuspec
-            yield (package, dependencies)
-            yield! buildDependencies dependencies
+            let dependenciesProjects = dependencies |> Set.toSeq |> Seq.collect (fun d -> d.Dependencies |> Set.toSeq) |> Seq.distinct
```
Comment (pchalamet) [inline src/FullBuild/Parsers/PackageRelationship.fs]: not sure to understand what dependencies of dependencies are in fact

File: src/FullBuild/Core/Simplify.fs [hunk 3]
```diff
@@ -160,11 +182,12 @@ let SimplifyAnthologyWithoutPackage (antho : Anthology) =
...
     newAntho
 
-let SimplifyAnthologyWithPackages (antho : Anthology) (package2files : Map<PackageId, AssemblyId set>) (package2packages : Map<PackageId, PackageId set>) =
+let SimplifyAnthologyWithPackages (antho : Anthology) (package2files : Map<PackageId, AssemblyId set>) (package2packages : Map<PackageId, PackageWithDependencies>) =
+    let package2packagesIgnoreFramework = package2packages |> Map.map (fun projectId dependencies -> dependencies.Dependencies |> Seq.collect(fun d-> d.Dependencies) |> Set.ofSeq)
     let newProjects = antho.Projects |> TransformSingleAssemblyToProjectOrPackage  package2files
                                      |> TransformPackageToProject
-                                     |> TransformPackagesToProjectsAndPackages package2packages package2files
-                                     |> RemoveAssembliesFromPackagesOrProjects package2files
+                                     |> TransformPackagesToProjectsAndPackages package2packagesIgnoreFramework package2files
```
Comment (pchalamet) [inline src/FullBuild/Core/Simplify.fs]: package2packagesIgnoreFramework is wrong so packages are not correctly removed then.

**Review by pchalamet — COMMENTED** [review]
fix package dependencies analysis

**Comment by blew** [comment]
Just wanted to cheer you guys on. We love you and really hope to be saved from this dreadful issue.

**Comment by pchalamet** [comment]
It will be fixed by the end of the week. No pressure ;-)

**Comment by pchalamet** [comment]
fixed in master - the bug was really elsewhere.

**Comment by blew** [comment]
still get the following error when run the fb convert command:
Error:
Missing repositories for indexation

**Comment by pchalamet** [comment]
That's something else and not related.
Just add other specified repository for conversion.

**Comment by pchalamet** [comment]
PR is closed - create another issue if you want to talk about that.

**Pull request closed without merging**